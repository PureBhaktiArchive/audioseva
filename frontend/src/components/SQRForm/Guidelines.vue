<template>
  <v-expansion-panels class="pt-3">
    <v-expansion-panel>
      <v-expansion-panel-header class="pl-0">Guidelines</v-expansion-panel-header>
      <v-expansion-panel-content>
        <slot></slot>
      </v-expansion-panel-content>
    </v-expansion-panel>
  </v-expansion-panels>
</template>

<script lang="ts">
import { Component, Vue } from "vue-property-decorator";

@Component({
  name: "Guidelines"
})
export default class GuideLines extends Vue {}
</script>

<style scoped>
</style>
