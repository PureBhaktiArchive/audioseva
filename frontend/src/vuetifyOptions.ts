export default {
  icons: {
    iconfont: 'fa',
    values: {
      plus: 'fas fa-plus',
      listening: 'fas fa-headphones',
      track: 'fas fa-cut',
      sound: 'fas fa-music',
      download: 'fas fa-file-download',
      delete: 'fas fa-trash',
      event: 'fas fa-calendar',
      undo: 'fas fa-undo',
      upload: 'fas fa-upload',
      check: 'fas fa-check',
    },
  },
} as any;
