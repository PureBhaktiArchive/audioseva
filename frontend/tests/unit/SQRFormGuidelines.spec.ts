import { mount, createLocalVue } from "@vue/test-utils";
import Vuetify from "vuetify";
import vuetifyOptions from "@/vuetifyOptions";
import Guidelines from "@/components/SQRForm/Guidelines.vue";

const localVue = createLocalVue();

describe("SQRFormGuidelines", () => {
  it("should render", () => {
    const wrapper = mount(Guidelines, {
      localVue,
      vuetify: new Vuetify(vuetifyOptions),
      slots: { default: "Guidelines text here" }
    });
    wrapper.find("button").trigger("click");
    expect(wrapper.html()).toMatchSnapshot();
  });
});
