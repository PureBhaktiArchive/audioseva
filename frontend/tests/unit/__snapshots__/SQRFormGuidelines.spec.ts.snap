// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SQRFormGuidelines should render 1`] = `
<div class="pt-3 v-item-group theme--light v-expansion-panels">
  <div aria-expanded="false" class="v-expansion-panel"><button type="button" class="v-expansion-panel-header pl-0">Guidelines<div class="v-expansion-panel-header__icon"><i aria-hidden="true" class="v-icon notranslate fas fa-chevron-down theme--light"></i></div></button>
    <div class="v-expansion-panel-content" style="display: none;">
      <div class="v-expansion-panel-content__wrap">Guidelines text here</div>
    </div>
  </div>
</div>
`;
